### DISCUSSION TITLE
Roadmap discussion

### DISCUSSION THREAD

**Discussion by aTrotier** [MagneticResonanceImaging/MRIReco.jl#149] [General]
Hello !

Since our last online meeting, we have implemented :
- A literate example for advanced reconstruction pipeline (binning data, extract kspace from acq and recreate acq struct from that) : **works offline but does not show the image in the documentation...** @aTrotier 
- noise decorrelation @Laura2305 
- Add different num of profiles along contrasts @aTrotier 
- add pogm/optista solver to RegularizedLeastSquares.jl @andrewwmao @JeffFessler 
- Geometric Coil Compression @aTrotier 
- Dedicated package for MRIFieldmaps : @JeffFessler @alexjaffray 
and of course a lot of bug fixes 👍 

Do you think we should work on a roadmap to focus our efforts ?
At least to know what we are working on or what we would like to see in the package.

## Roadmap suggestion :
- Fixing the accuracy / convergence issues for CS #114 
- implement and benchmark wavelet transformed for non-square / not 2^n matrix  https://github.com/tknopp/SparsityOperators.jl/issues/12
- Model-based reconstruction (moba in BART, PyQMRI)
- unified EPG package. for now we have multiple implementations in MRIReco and DECAES (other language implementation [sycomore](https://sycomore.readthedocs.io/en/latest/index.html) / EPGx)    @jondeuce / @cncastillo / @JakobAsslaender

**Comment by cncastillo** [comment]
I just wanted to say that I will soon implement a GPU-accelerated EPG simulator (using `KomaMRICore`). I will start trying to match [sycomore](https://sycomore.readthedocs.io/en/latest/index.html) in terms of functionality, so without the exchange and magnetization transfer (EPG-X) in the first iteration. I am interested in this to generate dictionaries for MRF reconstruction. Can you point me to the implementation in MRIReco?

**Comment by aTrotier** [comment]
Hi,
I have a simple epg implementation that mimic the sycomore implementation
(for now only the regular case) and can be used with automatic
differentiation : EPGsim.jl

In contrary to the implementation in MRI simulation.jl, you don't need to
define the whole epgstates array.

I did not benchmark which implementation is the fastest one. I guess it
will depends of the simulation.

**Comment by tknopp** [comment]
Hey @cncastillo and @aTrotier,

it would be great if you could unify the existing code/packages into MRISimulation: https://github.com/MagneticResonanceImaging/MRIReco.jl/tree/master/MRISimulation

If you prefer, the package could also be developed outside of MRIReco.jl. Would just be great to have it in the MagneticResonanceImaging organization.

**Comment by aTrotier** [comment]
@cncastillo if you want I can move my project into the organization, I don't know if you can use that for a gpu implementatio.

The only requirement on my part is to be able to use AD for sequence optimization with Cramer-Rao Lower Bound. Maybe @JakobAsslaender also have some suggestions about that.
And I have another package that used it for T2 mapping of a MultiEcho Spin Echo sequence that use it. I am style trying to figure out how to reduce the time to fit an epg curve : https://github.com/aTrotier/QuantitativeMRI.jl

By the way, the maintainer of sycomore is one of my collaborator. We can ask him about the epg implementation details

@tknopp What are your thoughts about having a dedicated epg package and that MRISimulation only used them to create the signal for specific sequence ?

**Comment by aTrotier** [comment]
[EPGsim.jl](https://github.com/MagneticResonanceImaging/EPGsim.jl) is now stored in the MagneticResonanceImaging organization

By the way @StevenWhitaker also has a package : https://github.com/StevenWhitaker/BlochSim.jl that also can be used for MR simulation

**Reply by tknopp** [reply]
I am not so deep into MRI simulation but I wonder if there would be the possibility to merge some of these packages into a single one.

**Reply by aTrotier** [reply]
I don't know if we should bring all of them into one package because they are not using the same mathematical/physics background. 
But I suppose we can try to have at least 2 main packages for EPG and Bloch Simulation for the community that at implement common functionality of the following packages

Packages, I am aware of, that use MR simulation :
Extended Phase Graph implementation :
- https://github.com/jondeuce/DECAES.jl
- https://github.com/MagneticResonanceImaging/EPGsim.jl
- MRSimulation.jl

Bloch simulation :
- https://github.com/StevenWhitaker/BlochSim.jl 
- https://github.com/JakobAsslaender/MRIgeneralizedBloch.jl
- https://github.com/cncastillo/KomaMRI.jl
- https://github.com/SpinDoctorMRI/SpinDoctor.jl

**Reply by aTrotier** [reply]
Another EPG implementation from @felixhorger : https://github.com/felixhorger/MRIEPG.jl
